File: social-service/src/main/java/com/app/socialservice/block/application/repository/BlockRepository.java
package com.app.socialservice.block.application.repository;

import com.app.socialservice.block.domain.model.Block;

import java.util.List;
import java.util.Optional;
import java.util.Set;
import java.util.UUID;

public interface BlockRepository {
    Optional<Block> findByUsers(UUID blockerUserId, UUID blockedUserId);
    Set<UUID> findBlockedUserIds(UUID userId);
    List<UUID> findBlockedUserIdsByBlockerId(UUID blockerUserId, int page, int limit);
    boolean existsByUsers(UUID blockerUserId, UUID blockedUserId);
    boolean insertIfAbsent(Block block);
    boolean deleteByUsers(UUID blockerUserId, UUID blockedUserId);
}

File: recommendation-service/src/block/repository/block_repository.py

from abc import ABC, abstractmethod
from uuid import UUID

from block.model.block import Block


class BlockRepository(ABC):
    @abstractmethod
    def create_block(self, block: Block) -> None:
        pass

    @abstractmethod
    def remove_block(self, blocker_user_id: UUID, blocked_user_id: UUID) -> None:
        pass
